### ISSUE TITLE
Mapping for iframe is wrong

### ISSUE THREAD

**Issue by minorninth** [w3c/html-aam#130]
For the Mac AX API in particular, the iframe element does not map to AXWebArea. Rather, the iframe element maps to a generic / unknown role (AXUnknown) and is hidden from the accessibility tree, and the child/descendant of the iframe is the one that has the role of AXWebArea.

**Comment by minorninth** [comment]
I propose something like this:

AXRole: not mapped
Child: AXRole of AXWebArea, AXRoleDescription of "html content"

**Comment by medeaoblongata** [comment]
We've had a lot of problems with our auditors expecting iframes to have accessible names. 

I can imagine situations when they **should** be named, and situations when they need not. Clearly the spec is silent on this, but because of the way real world ATs work with iframes – they are invariably announced in their own right – names seem to be expected.

In our case, naming every iframe in the DOM hierarchy risks causing a lot of duplicate announcements because the LMS that delivers our content uses **two** nested iframes before our code even gets a look-in. Since that LMS is maintained by another team, and another firm, It has proved incredibly difficult to get those elements to have a meaningful name, because it's "just" DOM architecture (i.e. I want them to have role="generic"). 

Similarly, we would love to be able to hoist the correct lang attribute beyond our content to the container iframe(s), but this has proved almost impossible because we do not control the LMS. The language of the content is not "known" by the LMS.

Unfortunately it's **our** content that gets dinged with WCAG violations. (Either "Page Titled", "Name, Role, Value", or "Headings and Labels", or in the lang case "Language of Page" or "Language of Parts"). Infuriating.

Clarification and/or guidance on this point (name required or not) would be very much appreciated. Also, how might we mark an iframe as "purely structural" given that role="generic" is not for authors' use? 

Would it not be most appropriate for iframes to have generic role by default (but not hiding the content inside from the A11y tree), and if the iframes are semantically meaningful elements requiring names, authors may add role="section" or role="complementary"?

**Comment by scottaohara** [comment]
@brennanyoung it sounds more like you have an issue with how people are interpreting wcag.  

it wouldn't be a good idea to give iframes generic roles, because that's not what they are. this issue is about correctly identifying the existing platform role that iframes have.  and the ARIA issue you commented on is about determining if there needs to be a unified ARIA role, instead of the disparate platform roles.  (likely doesn't at this point)
